RecoMTD/DetLayers/interface/MTDDetLayerGeometry.h

```diff
@@ -0,0 +1,82 @@
+#ifndef DetLayers_MTDDetLayerGeometry_h
+#define DetLayers_MTDDetLayerGeometry_h
+
+/** \class MTDDetLayerGeometry
+ *
+ *  Provide access to the DetLayers of mip timing detectors.
+ *
+ *  \author L. Gray - FNAL
+ *  
+ */
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include "TrackingTools/DetLayers/interface/DetLayerGeometry.h"
+#include <vector>
+#include <map>
+
+class DetLayer;
+
+class MTDDetLayerGeometry : public DetLayerGeometry{
+ public:
+
+  /// Constructor
+  MTDDetLayerGeometry();
+
+  friend class MTDDetLayerGeometryESProducer;  
+
+  /// Destructor
+  ~MTDDetLayerGeometry() override;
+
+  /// return all barrel layers
+  const std::vector<const DetLayer*>& allBarrelLayers() const;
+
+  /// return all endcap layers
+  const std::vector<const DetLayer*>& allEndcapLayers() const;
+
+  /// return all endcap layers
```

Comment:
do all of these actually return all endcap layers, or do the forward and backward ones return something else?

Reply:
It's all the endcap layers (forward and backward) ordered in Z.